packages/fetchai/connections/soef/connection.py

```diff
@@ -1286,21 +1292,29 @@ def __init__(self, **kwargs: Any) -> None:
                 self.DEFAULT_CONNECTION_CHECK_MAX_RETRIES,
             ),
         )
+        is_https = cast(bool, self.configuration.config.get("is_https"))
         soef_addr = cast(str, self.configuration.config.get("soef_addr"))
         soef_port = cast(int, self.configuration.config.get("soef_port"))
         chain_identifier = cast(str, self.configuration.config.get("chain_identifier"))
         token_storage_path = cast(
             Optional[str], self.configuration.config.get("token_storage_path")
         )
-        if api_key is None or soef_addr is None or soef_port is None:  # pragma: nocover
-            raise ValueError("api_key, soef_addr and soef_port must be set!")
-
+        not_none_params = {
+            "api_key": api_key,
+            "soef_addr": soef_addr,
+            "soef_port": soef_port,
+        }
+        for param_name, param_value in not_none_params.items():  # pragma: nocover
+            if param_value is None:
+                raise ValueError(f"{param_name} must be set!")
         self.api_key = api_key
+        self.is_https = is_https if is_https is not None else True
```

Comment:
None Pythonic, also see above we can set default

Reply:
```suggestion
        self.is_https = is_https
```